runners/core-construction-java/src/main/java/org/apache/beam/runners/core/construction/resources/PipelineResourcesDetector.java

```diff
@@ -18,10 +18,10 @@
 package org.apache.beam.runners.core.construction.resources;
 
 import java.io.Serializable;
-import java.util.List;
+import java.util.stream.Stream;
 
 /** Interface for an algorithm detecting classpath resources for pipelines. */
 public interface PipelineResourcesDetector extends Serializable {
 
-  List<String> detect(ClassLoader classLoader);
+  Stream<String> detect(ClassLoader classLoader);
```

Comment:
Would prefer `List`.

Reply:
Stream was a better fit in case we want to do filtering outside a `PipelineResourcesDetector` implementation class. This is already done in `PipelineResources.java`. This way, when a detector (even a 3rd party one) provides paths that we know we should not accept, we can easily filter them out using stream's api and then materialize the result to a collection of our choice (eg. List or Set - it's up to beam's developers at this point).

Reply:
> I would have also preferred list since stream can imply that it is infinitely long since there could be a generator function the person implements.

This is a good point - I didn't think about that from that perspective. Thanks!